### PR TITLE
fix:  misused `max_priority_fee_per_gas` and `max_fee_per_gas`

### PR DESCRIPTION
**Pull request by dyxushuai**
## Description

According to the description of [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) and the definition in Geth 
https://github.com/ethereum/go-ethereum/blob/67979022aa8ea6a96b618c086f60b49af7d2a568/core/types/tx_dynamic_fee.go#L29-L30
```go
        GasTipCap  *big.Int // a.k.a. maxPriorityFeePerGas
        GasFeeCap  *big.Int // a.k.a. maxFeePerGas
```
So, the origin codes misused the mapping

 
## Issue Link

No

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
Hi @dyxushuai, please add the PR description. It should describe the motivation behind the change.

**Comment by dyxushuai** [comment]
@ChihChengLiang I have made the changes. Please look over once more.

File: eth-types/src/geth_types.rs [hunk 1]
```diff
@@ -180,8 +180,8 @@ impl From<&Transaction> for crate::Transaction {
             gas: tx.gas_limit.to_word(),
             value: tx.value,
             gas_price: Some(tx.gas_price),
-            max_priority_fee_per_gas: Some(tx.gas_fee_cap),
-            max_fee_per_gas: Some(tx.gas_tip_cap),
+            max_priority_fee_per_gas: Some(tx.gas_tip_cap),
+            max_fee_per_gas: Some(tx.gas_cap_cap),
```
Comment (ChihChengLiang) [inline eth-types/src/geth_types.rs:184]:
```suggestion
            max_fee_per_gas: Some(tx.gas_fee_cap),
```

File: eth-types/src/geth_types.rs [hunk 2]
```diff
@@ -201,8 +201,8 @@ impl From<&crate::Transaction> for Transaction {
             gas_limit: tx.gas.as_u64().into(),
             value: tx.value,
             gas_price: tx.gas_price.unwrap_or_default(),
-            gas_fee_cap: tx.max_priority_fee_per_gas.unwrap_or_default(),
-            gas_tip_cap: tx.max_fee_per_gas.unwrap_or_default(),
+            gas_tip_cap: tx.max_priority_fee_per_gas.unwrap_or_default(),
+            gas_cap_cap: tx.max_fee_per_gas.unwrap_or_default(),
```
Comment (ChihChengLiang) [inline eth-types/src/geth_types.rs:205]:
```suggestion
            gas_fee_cap: tx.max_fee_per_gas.unwrap_or_default(),
```

**Review by ChihChengLiang — CHANGES_REQUESTED** [review]
Thanks for updating the PR description. It now makes more sense.
I noticed typos on variables. Can you fix them?

**Comment by dyxushuai** [comment]
> Thanks for updating the PR description. It now makes more sense. I noticed typos on variables. Can you fix them?

Sorry, I edited it in browser without testing. Thank you for pointing out

**Review by ChihChengLiang — APPROVED** [review]
LGTM. Thank you for the fix!

**Review by hero78119 — APPROVED** [review]
LGTM! thanks for the fix

**Pull request merged**